packages/package_info/package_info_web/lib/package_info_web.dart

```diff
@@ -0,0 +1,59 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:http/http.dart';
+
+///Web Implementation of PackageInfo
+class PackageInfoPlugin {
+  ///Register with Registrar
+  static void registerWith(Registrar registrar) {
+    final MethodChannel channel = MethodChannel(
+        'plugins.flutter.io/package_info',
+        const StandardMethodCodec(),
+        registrar.messenger);
```

Comment:
Web plugins normally use a federated approach to their implementation, instead of registering a Method Channel (there's no need for a Method Channel when all the code that runs is Dart).

Could you please federate `package_info` properly, then implement the web version as an implementation of that federation? That'll not only enable a cleaner web implementation, but also allow for other platforms to have their own version as well!

Thanks!

Reply:
I have several doubts regarding the federated_plugin approach . First the only method the package_info plugin has is a static method . so how will the other plugin override it. And the fromPlatform() method returns an instance of itself. so how would i create a federated_plugin without making breaking changes.